addons/purchase_stock/tests/test_stockvaluation.py

```diff
@@ -1142,17 +1137,23 @@ def _now(*args, **kwargs):
         full_reconcile = stock_lines.mapped('full_reconcile_id')
         self.assertTrue(full_reconcile.exists())
 
-        reconciled_lines = full_reconcile.reconciled_line_ids - stock_lines
-        self.assertEqual(len(reconciled_lines), 2)
+        reconciled_lines = full_reconcile.reconciled_line_ids
+        self.assertRecordValues(full_reconcile.reconciled_line_ids, [
+            # Exchange difference lines:
+            {'balance': -185.72, 'amount_currency': 0.0},
+            {'balance': 232.15, 'amount_currency': 0.0},
+            {'balance': 92.86, 'amount_currency': 0.0},
+            # Other lines:
+            {'balance': 50.0, 'amount_currency': 100.0},
+            {'balance': 25.0, 'amount_currency': 50.0},
+            {'balance': -214.29, 'amount_currency': -150.0},
+        ])
+        self.assertEqual(len(reconciled_lines), 6)
 
         stock_journal_line = reconciled_lines.filtered(lambda l: l.journal_id == self.stock_journal)
         self.assertEqual(stock_journal_line.amount_currency, -150)
         self.assertAlmostEqual(stock_journal_line.balance, -214.29)
 
-        exchange_stock_line = reconciled_lines.filtered(lambda l: l.journal_id == exchange_diff_journal)
-        self.assertEqual(exchange_stock_line.amount_currency, 0.00)
-        self.assertAlmostEqual(exchange_stock_line.balance, 139.29)
```

Comment:
so here, the value of 139.29 can't be found anymore since we compute the exchange difference from the amount in foreign currency instead of the company currency, right?

Reply:
Indeed. Also, I added an assert above on the full `reconciled_line_ids` recordset instead of this minimal assert